### PR TITLE
Upgraded dependency to newest versions and corrected some tests

### PR DESCRIPTION
**Pull request by ChrisRo89**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@ under the License.
   </scm>
 
   <properties>
-    <revision>1.1.3</revision>
+    <revision>1.1.4</revision>
```
Comment (kuisathaverat) [inline pom.xml]: this version number should change only when we release the plugin

File: src/main/java/org/jenkinsci/plugins/saml/IdpMetadataConfiguration.java [hunk 2]
```diff
@@ -179,7 +180,7 @@ public String getDisplayName() {
             return "";
         }
 
-        public FormValidation doTestIdpMetadata(@QueryParameter("xml") String xml) {
+        public FormValidation doTestIdpMetadata(@QueryParameter("xml") String xml) throws IOException {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/IdpMetadataConfiguration.java]: it should return a Fromvalidation with an error message, it never should throw an exception

File: src/main/java/org/jenkinsci/plugins/saml/SamlEncryptionData.java [hunk 3]
```diff
@@ -31,6 +31,7 @@
 import org.apache.commons.lang.StringUtils;
 import org.kohsuke.stapler.QueryParameter;
 
+import java.io.Console;
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlEncryptionData.java]: it seems not needed

File: src/main/java/org/jenkinsci/plugins/saml/SamlFileResource.java [hunk 4]
```diff
@@ -61,11 +65,36 @@ public boolean exists() {
...
+    }
+
     @Override
     public String getFilename() {
         return fileName;
     }
 
+    @Override
+    public String getDescription() {
+        return null;
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlFileResource.java]: maybe return the name of the file instead to return a null

File: src/main/java/org/jenkinsci/plugins/saml/SamlValidateIdPMetadata.java [hunk 5]
```diff
@@ -40,7 +43,7 @@ protected FormValidation process() {
             return FormValidation.error("The IdP Metadata not valid.", e);
         } catch (XMLParserException e) {
             return FormValidation.error("The IdP Metadata not valid XML.", e);
-        } catch (ComponentInitializationException e) {
+        }  catch (ComponentInitializationException e) {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlValidateIdPMetadata.java]: nit

File: src/test/java/org/jenkinsci/plugins/saml/SamlFormValidationsTest.java [hunk 6]
```diff
@@ -73,132 +73,132 @@ public void start() {
     @LocalData("testReadSimpleConfigurationAdvancedConfiguration")
     @Test
     public void testIdpMetadata() throws Exception {
-        assertEquals(idpMCdescriptor.doTestIdpMetadata(null).kind, Kind.ERROR);
-        assertEquals(idpMCdescriptor.doTestIdpMetadata("").kind, Kind.ERROR);
-        assertEquals(idpMCdescriptor.doTestIdpMetadata(" ").kind, Kind.ERROR);
+        assertEquals(Kind.ERROR, idpMCdescriptor.doTestIdpMetadata(null).kind );
```
Comment (kuisathaverat) [inline src/test/java/org/jenkinsci/plugins/saml/SamlFormValidationsTest.java]: I guess that inverts all the assertEquals is for a better read of the asserts but it is not related to the change, it uses to be in another PR, but it is fine.

File: src/main/java/org/jenkinsci/plugins/saml/SamlSPMetadataWrapper.java [hunk 7]
```diff
@@ -39,8 +41,8 @@ public SamlSPMetadataWrapper(SamlPluginConfig samlPluginConfig, StaplerRequest r
      * @throws IllegalStateException if something goes wrong.
      */
     @Override
-    protected HttpResponse process() throws IllegalStateException {
+    protected HttpResponse process() throws IllegalStateException, IOException {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSPMetadataWrapper.java]: new IOException

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 8]
```diff
@@ -244,7 +245,7 @@ public String getLoginUrl() {
      * @return the http response.
      */
     public HttpResponse doCommenceLogin(final StaplerRequest request, final StaplerResponse response, @QueryParameter
-            String from, @Header("Referer") final String referer) {
+            String from, @Header("Referer") final String referer) throws IOException {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: new IOException

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 9]
```diff
@@ -292,7 +293,7 @@ private String calculateSafeRedirect(String from, String referer) {
      * @return the http response.
      */
     @RequirePOST
-    public HttpResponse doFinishLogin(final StaplerRequest request, final StaplerResponse response) {
+    public HttpResponse doFinishLogin(final StaplerRequest request, final StaplerResponse response) throws IOException {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: new IOException

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 10]
```diff
@@ -568,7 +569,7 @@ static String getSPMetadataFilePath() {
      * @param response http response.
      * @return the http response.
      */
-    public HttpResponse doMetadata(StaplerRequest request, StaplerResponse response) {
+    public HttpResponse doMetadata(StaplerRequest request, StaplerResponse response) throws IOException {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: new IOException

File: src/main/java/org/jenkinsci/plugins/saml/SamlFileResource.java (hunk 4 above)
Comment (ChrisRo89) [reply src/main/java/org/jenkinsci/plugins/saml/SamlFileResource.java]: Thank you, will change it

File: pom.xml [hunk 11]
```diff
@@ -77,6 +77,16 @@ under the License.
       <id>repo.jenkins-ci.org</id>
       <url>https://repo.jenkins-ci.org/public/</url>
     </repository>
+    <repository>
+      <id>shib-release</id>
+      <url>https://build.shibboleth.net/nexus/content/groups/public</url>
```
Comment (recena) [inline pom.xml]: @ChrisRo89 Why? Is there any dependency available exclusively on this repository?

Comment (recena) [reply pom.xml]: `net.shibboleth.utilities`? not in Maven Central?

File: src/main/java/org/jenkinsci/plugins/saml/IdpMetadataConfiguration.java (hunk 2 above)
Comment (ChrisRo89) [reply src/main/java/org/jenkinsci/plugins/saml/IdpMetadataConfiguration.java]: Thank you for the hint, I changed it

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 12]
```diff
@@ -578,8 +581,14 @@ static String getSPMetadataFilePath() {
      * @param response http response.
      * @return the http response.
      */
-    public HttpResponse doMetadata(StaplerRequest request, StaplerResponse response) throws IOException {
-        return new SamlSPMetadataWrapper(getSamlPluginConfig(), request, response).get();
+    public HttpResponse doMetadata(StaplerRequest request, StaplerResponse response) {
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: return null it is a bad idea, the IOException is fine here

**Comment by kuisathaverat** [comment]
Could you open a Jira with the details about this PR, why and what we want to achieve here, then add the link to the first comment of the PR see how on other PRs https://github.com/jenkinsci/saml-plugin/pulls?q=is%3Apr+is%3Aclosed

File: pom.xml [hunk 13]
```diff
@@ -150,6 +195,223 @@ under the License.
       <version>2.0.0.0</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 14]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>jcl-over-slf4j</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>commons-codec</groupId>
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.springframework</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: wrong dependency pac4j-saml uses spring-core 5.0.4

File: pom.xml [hunk 15]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>jcl-over-slf4j</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 16]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+        </exclusion>
+        <exclusion>
+          <groupId>commons-codec</groupId>
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
+      <artifactId>opensaml-messaging-impl</artifactId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 17]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>jcl-over-slf4j</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>commons-codec</groupId>
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 18]
```diff
@@ -150,6 +195,223 @@ under the License.
...
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 19]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 20]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>jcl-over-slf4j</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 21]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 22]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 23]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+      <artifactId>opensaml-security-impl</artifactId>
+      <version>3.4.3</version>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need 3.4.0 is implicit in pac4j-saml

File: pom.xml [hunk 24]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+          <artifactId>jcl-over-slf4j</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>commons-codec</groupId>
+          <artifactId>commons-codec</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.santuario</groupId>
```
Comment (kuisathaverat) [inline pom.xml]: not need

File: pom.xml (hunk 11 above)
Comment (kuisathaverat) [reply pom.xml]: opensaml >3.3x is not in the Maven Central, the latest version in Maven Central is 3.3.0 from 2017, it seems related to security reasons https://wiki.shibboleth.net/confluence/display/DEV/Use+of+Maven+Central

File: pom.xml [hunk 25]
```diff
@@ -150,6 +195,223 @@ under the License.
...
+        </exclusion>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-api</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>org.opensaml</groupId>
+      <artifactId>opensaml-security-api</artifactId>
```
Comment (kuisathaverat) [inline pom.xml]: not needed 3.4.0 is implicit in pac4j-saml

File: pom.xml (hunk 14 above)
Comment (kuisathaverat) [reply pom.xml]: and here is the big problem the Jenkins core has spring 2.5.6, so add a dependency to spring 5.0.4 is a highly risky decision that affects other plugins and Jenkins Core, I've remembered that was was the cause I did not move to pac4j 2.0.0 I have plans to move everything to use openSAML without pac4j but it will not be soon.
